bluetooth: controller: RX PDU meta data support in LLL

mtpr-ot opened the pull request:
Added support for vendor specific meta data in LLL node_rx_hdr. This
enables vendors to add "footer" data to the RX PDU, for supporting
specialized BLE features.

Signed-off-by: Morten Priess <mtpr@oticon.com>

cvinayak commented on subsys/bluetooth/controller/Kconfig:
```diff
@@ -535,6 +535,10 @@ config BT_CTLR_LOW_LAT_ULL
 	  loop to demux rx messages from LLL.
 
+config BT_CTLR_RX_PDU_META
+	prompt "Enable proprietary RX pdu meta data"
```
remove "proprietary"

cvinayak commented on subsys/bluetooth/controller/hci/hci.c:
```diff
@@ -27,6 +27,9 @@
 #include "hal/ccm.h"
 #include "ll_sw/pdu.h"
+#if defined(CONFIG_BT_CTLR_RX_PDU_META)
+#include "lll_ext.h"
```
rename to lll_meta.h  and include in lll.h only. (less number of file changes)

mtpr-ot committed:
bluetooth: controller: RX PDU meta data support in LLL

Added support for vendor specific meta data in LLL node_rx_hdr. This
enables vendors to add "footer" data to the RX PDU, for supporting
specialized BLE features.

Signed-off-by: Morten Priess <mtpr@oticon.com>

cvinayak reviewed (approved):
LGTM. The added Kconfig option does not have a help description, but this is fine for now as a place holder until vendors to define the meta data structure.

mtpr-ot commented:
@aescolar, could you please merge? Thanks.
